src/main/java/org/takes/facets/fork/FkTypes.java

```diff
@@ -51,25 +52,55 @@
     /**
      * Response to return.
      */
-    private final transient Response origin;
+    private final transient Opt<Response> response;
+
+    /**
+     * Response to return.
+     */
+    private final transient Opt<Take> take;
+
+    /**
+     * Ctor.
+     * @param list List of types
+     * @param resp Response to return
+     */
+    public FkTypes(final String list, final Response resp) {
+        this(list, new Opt.Single<>(resp), new Opt.Empty<Take>());
+    }
+
+    /**
+     * Ctor.
+     * @param list List of types
+     * @param tke The take to use to build the response to return
+     */
+    public FkTypes(final String list, final Take tke) {
```

Comment:
@essobedo please describe in PR description why do we need to touch `FkTypes` and why we need 2 new constructors